### DISCUSSION TITLE
(Auto)playlists navigation

### DISCUSSION THREAD

**Discussion by regorxxx** [TT-ReBORN/Georgia-ReBORN#83] [Ideas]
Hiperlinks create new playlists with queries, but there is no way to go back. It would be treat to:

- Be able to go back to previous playlist from the "search" playlist. This would require a new back button somewhere on the UI to go to previously visited playlist.
- Create undo backups points before loading tracks on Search playlist. That way pressing ctrl + z would allow to load previous search.

**Comment by TT-ReBORN** [comment]
Implemented and will be available in Beta 10d. When a hyperlink was clicked and the "Search" playlist created, there will be the option "Back to previous playlist" in the playlist context menu, in the playlist manager and in top menu Playlists.

Glad you mentioned this @regorxxx =)

**Reply by kbuffington** [reply]
Can you link me to a diff? I'm very curious how this was implemented.

**Reply by TT-ReBORN** [reply]
Look who is here! Kevin... I've missed you :*( 
What have you been doing man!?

Can't show you Github commits now, because well... I've been doing too much and Github activity has been stopped till I release the first public release. After that everything on Github will be normal again.

The implementation was actually really easy and nothing fancy about it. 
I've uploaded it for you to test, [here is the next WIP Beta 10d that has the feature](https://cloud.multibits.eu/s/A8qYqtZ9jeFFE63). 
I don't know if you ever installed my theme, but basically you just do a portable installation and drag the profile folder from the zip into foobar's root dir. Start foobar and add the Library. Then just click on a hyperlink in the playlist which will create the "Search" playlist, right click for playlist context menu > Back to previous playlist.

Code = line 19, 960, 4737 or just search "lastActivePlaylist" and you will see.

-Tom

**Reply by kbuffington** [reply]
I'm still alive. I've got plans for cool shit to implement in Georgia, but haven't had time after switching jobs, having surgery, working on other projects, etc., so I've kinda let things slide on this front unfortunately. I'll give the beta a look-see soon!

**Reply by TT-ReBORN** [reply]
OK! I was worried you lost interest, glad it didn't turn that way ;-)

You really should try the beta because I've done so much work, it's a completely new theme. You will be also interested in the new code and structure! It's not using lodash anymore and not even one var :-) . Btw when you use the new beta, try double clicking on the lower bar. I have added really cool stuff, you should also check all new features in top menu Options.
More info also in the changelog -> foobar2000\profile\georgia-reborn\docs\CHANGELOG.md changes since the Github version to the Beta version or all the changelogs in the zip file.

I would love if you could write a component that auto-fetches disc art from fanart.tv, I am certain all users would love 
and appreciate it!

-Tom

**Comment by kbuffington** [comment]
So I _still_  haven't downloaded your latest beta yet (I will I swear!) but I was stuck in a house without wifi over thanksgiving, and decided to just implement playlist history. Thought you might find this commit interesting: https://github.com/kbuffington/Georgia/commit/47b75362933c2a4cf37e5af57ef8679ab2536348

<img width="955" alt="image" src="https://user-images.githubusercontent.com/2282004/204403580-cf7a0c1a-3336-4cfe-8ebe-ec4edec84a08.png">

Back/Forward arrows are enabled when possible, just like a browser window, and it allows 10 history states to be saved as you're clicking around. States stored in the history are: playlist switches, song reorders, and additions/deletions, and it works just fine with autoplaylists. Adding or deleting a playlist unfortunately has to reset the playlist history.

**Reply by TT-ReBORN** [reply]
Thanks Kevin! It's much better than my "1 step back when clicking on hyperlinks" implementation.

But the big question is that "you were stuck in a house", that sounds funny. 
So I guess you visited some family/mother in law and you got tired/bored after a while, so you've put up your laptop in the living room, coding this and eating some baked cookies from your mother in law while appreciating her cooking skills? 
Love it ;-)

-Tom

**Reply by kbuffington** [reply]
Haha, we rented a big AirBNB with a bunch of family near the beach, but it was cold and rainy most of the time, so I had plenty of time while watching football games.

That code, particularly the UI part, isn't perfect yet (on 4k screens at least), but the actual PlaylistHistory stuff seems good. Let me know if you find any bugs.
